arch/arm/src/stm32h5/stm32_adc.c

```diff
@@ -624,6 +773,195 @@ static void adc_wdog_enable(struct stm32_dev_s *priv)
   adc_putreg(priv, STM32_ADC_IER_OFFSET, regval);
 }
 
+/****************************************************************************
+ * Name: adc_wdog1_configure
+ *
+ * Description:
+ *
+ ****************************************************************************/
+
+static int adc_wdog1_configure(struct stm32_dev_s *priv,
+                                struct stm32_adc_watchdog1_cfg_s *cfg)
```

Comment:
This should be generalized and be a single function for all ADC instances. There's no point in duplicating the logic.